system/HTTP/Message.php

```diff
@@ -140,7 +140,7 @@ public function getProtocolVersion(): string
 	 *
 	 * @deprecated Use header calls directly
 	 */
-	public function isJSON()
+	public function isJSON(): bool
```

Comment:
revert this
```suggestion
      public function isJSON()
```

Reply:
@samsonasik Just so I know in the future, can you tell me why this needed to be reverted? There are plenty of places where return types are being used in the framework and this function is meant to only return a boolean.

Reply:
change return type declaration is BC break https://3v4l.org/YHmT2

Reply:
@samsonasik The method must explicitly return a boolean value.
The method was added to the development branch between 4.0.4 and 4.1.0 (4.0.5) releases. So it cannot be BC. 
Your example is not a BC, but a bad practice implementation. 

What you did was a slap in the face. "I'm the boss. You're a fool "(с)

Reply:
The method marked as `@deprecated` so changing return type is not make sense.It may considered to be removed entirely instead.

The example is BC break, once we define public/protected method on non-final class, we need to maintain that.

Reply:
This method has been marked as ```@deprecated``` but has never been included in any release. It could just as easily have been removed. There is no reason to mark it as ```@depratacted```.

The framework core does not override this method, so BC cannot be.
I repeat. This method was not available until release 4.1.
But suddenly he began to break something. 

Stop justifying insanity!

Reply:
@iRedds could you provide a PR to remove this? Thank you.